view/theme/frio/js/theme.js

```diff
@@ -764,12 +764,52 @@ function htmlToText(htmlString) {
  * @param {boolean} un    Whether to perform an activity removal instead of creation
  */
 function doActivityItemAction(ident, verb, un) {
-	if (verb.indexOf("attend") === 0) {
-		$(".item-" + ident + " .button-event:not(#" + verb + "-" + ident + ")").removeClass("active");
-	}
-	$("#" + verb + "-" + ident).toggleClass("active");
+  console.log(ident, verb, un);
+	_verb = un ? 'un' + verb : verb;
+	$('#like-rotator-' + ident.toString()).show();
+	$.post('item/' + ident.toString() + '/activity/' + _verb)
+	.success(
+		function(data){
+      console.log("data.status: " + data.status);
+      if (data.status == "ok") {
+        console.log("connection: " + data.status);
+        $('#like-rotator-' + ident.toString()).hide();
+        if (verb.indexOf("announce") === 0 ) {
+          console.log("announce")
+          if (data.verb == "un" + verb) {
+            $("button[id^=shareMenuOptions-" + ident.toString() + "]" ).removeClass("active");
+            $("button[id^=shareMenuOptions-" + ident.toString() + "]" ).attr("onclick", "doActivityItemAction(" + ident +", '" + verb + "', " + false + ")").change();
```

Comment:
These structures can be simplified, please change the other cases.
```suggestion
            $("button[id^=shareMenuOptions-" + ident.toString() + "]" )
                .removeClass("active")
                .attr("onclick", "doActivityItemAction(" + ident +", '" + verb + "', " + false + ")").change();
```